### ISSUE TITLE
[Mobile] ScrollZone parallax does not take into account autofocus cropping

### ISSUE THREAD

**Issue by raineorshine** [cybersemics/em#2651]
## Steps to Reproduce

``` 
- A
- B
- C
- D
- E
- F
- G
- H
- I
- J
- K
- L
- M
- N
- O
- P
- Q
- R
- S
- T
- U
- V
- X
- Y
- Z
  - 1
    - 2
``` 

Move the cursor between `Z` and `1`. 

## Current Behavior

The scroll zone background jumps. 

https://github.com/user-attachments/assets/64cadb28-5a29-4adb-9928-920e7d11c952

## Expected Behavior

The scroll zone background position should not move. It should not be affected by the autofocus cropping.

**Comment by snqb** [comment]
The only thing that came up to my mind that is not gonna be too complex is just not moving it for smaller jumps

Since the `ScrollZone` just listens for `y`, and then moves the bg by `y / 4 + 400` pixels, it's hard not to trigger the move.
A sophisticated and complex solution could mean trying to differentiate between autofocus and scroll, but do we really want that?
A simpler one could be just comparing latest Y positions and try to judge whether that's a focus thing or not.

**Comment by snqb** [comment]
@raineorshine on a side-note, I wasn't really planning to merge the bg thing, as I was looking for better backgrounds(ours was crudely cut imo)

I have this on my local branch right now, isn't it a bit prettier?
https://github.com/user-attachments/assets/dcc8e1a0-bff5-4c8f-8239-615fe75d595b


concerning the autofocus scroll, I'm looking into it

**Comment by snqb** [comment]
> No, there's no need to guess or approximate here. We already have the exact scroll amount that is jumped for autofocus, so we just need to adjust the parallax to use the autofocus-adjusted y value instead of the actual scrollY.

I think I've found a solution that allows us not to do calculations https://github.com/cybersemics/em/pull/2652

**Comment by snqb** [comment]
so, it should not move when we have autofocus crop(when we go from Z to 1), a level deeper
but should when we go J -> K or A -> B or D -> C by tapping the cursor

right?

**Comment by raineorshine** [comment]
I'm sorry, I don't understand. Can you clarify the question?

**Comment by snqb** [comment]
Let me rephrase it, yup.
So we don't want the background to move on "autofocus crop". It means Z -> 1 should not trigger any bg move.

But should we trigger a bg move if I switch from A to J on the other side of the screen?
Illustrating this:

https://github.com/user-attachments/assets/c9d30dc2-a3cd-4e5d-9a92-0e2a66589bcc

**Comment by raineorshine** [comment]
> So we don't want the background to move on "autofocus crop". It means Z -> 1 should not trigger any bg move.

Yes, that's right.

> But should we trigger a bg move if I switch from A to J on the other side of the screen?
> Illustrating this:

I'm not sure what "switch" means and I'm not sure what "the other side of the screen" means. I can't make out what's happening in the video. I can see the cursor is moving, but I'm not sure where you are touching or how it's getting there.

In general, parallax works by translating the scroll zone background proportionally to scrollY. However, when the user navigates to a deeper depth and the autofocus kicks in, the scroll position is artificially altered so that the empty space above the thoughts can be reduced. To the user, it appears that the scroll position has not changed, because the actual scroll position is translated by exactly the same amount that the empty space is reduced at the same time.

The task is to alter the parallax algorithm so that it is based on this perceived scroll position rather than the actual scroll position. I believe this difference is stored in a single variable in the LayoutTree, so the tricky part is just how to get it to the scroll zone cleanly.

**Comment by snqb** [comment]
I was clicking an item right on the bottom of the screen to illustrate the autoscroll, but nvm, I didn't have the right perspective.

The explanation is beatiful, thanks! I'm having a look

**Comment by snqb** [comment]
hey, I think I'm going into a rabbit hole on this.
I've checked lots of variables like spaceAbove, spaceBetween, diffes of Ys and things, and none to my eye correllated with the scroll change in a simple manner. And then also I'll have to find a not-that-complex-way to share it with  `ScrollZone`. Guess, I'm bad for this task  :D

but btw i have a dumb idea: what if we just turn off parallax for forced scrolls?
feels like this kind of thing would be simpler in all matters, and getting it into a scrollZone is just making `forcedScrolling` exported
it also feels like a more correct(dumb) way to handle this.
```typescript

import { forcedScrolling } from '../redux-middleware/scrollCursorIntoView'

/** An overlay for the scroll zone that blocks pointer events. */
const ScrollZone = ({ leftHanded }: { leftHanded?: boolean } = {}) => {
  const [scrollTop, setScrollTop] = useState(0)

  scrollTopStore.useEffect(s => {
    if (!forcedScrolling) setScrollTop(s)
  })
 //  .. . . . 
  return <div tranlateY={scrollTop} />
```

otherwise, let me just not charge for this, and move on with https://github.com/cybersemics/em/issues/2558, I think I know how to fix it easily

**Comment by raineorshine** [comment]
Thanks for trying, and sorry for the trouble! The autocropping is not the easiest logic. I'd like to give it a shot myself though before we look for other means.

The issue I have with disabling parallax is that it then becomes decoupled from the scroll position. This seems wrong to me. It should be a a deterministic function of the perceived scroll position. Consider how objects in the distance move proportionally slower than objects in the foreground. It is a physical property of perspective, so it shouldn't turn on and off. Another way to think about it: If scrollTop is 500, say, then the scroll zone background should be in the same place regardless of whether I got their programmatically or scrolled there myself.
